test-integ/peering_commontopo/ac5_2_test.go

```diff
@@ -0,0 +1,315 @@
+package peering
+
+import (
+	"fmt"
+	"time"
+
+	"testing"
+
+	"github.com/hashicorp/consul/api"
+	"github.com/hashicorp/consul/sdk/testutil/retry"
+	"github.com/hashicorp/consul/testingconsul/topology"
+	"github.com/stretchr/testify/require"
+)
+
+// This tests prepared query failover across 3 clusters. The following test case covers:
+// 1. Create a prepared query with failover target to DC2 and DC3 cluster.
+// Test will fail if not run in parallel
+type preparedQueryFailoverSuite struct {
+	DC   string
+	Peer string
+
+	nodeServer   []*topology.Node
+	dc3ServerSID topology.ServiceID
+
+	serverSID    topology.ServiceID
+	clientSID    topology.ServiceID
+	dc3APIClient *api.Client
+}
+
+var (
+	preparedQueryFailoverSuites []*preparedQueryFailoverSuite = []*preparedQueryFailoverSuite{
+		{DC: "dc1", Peer: "dc2"},
+		{DC: "dc2", Peer: "dc1"},
+	}
+	query *api.PreparedQuery
+	def   *api.PreparedQueryDefinition
+)
+
+func TestPreparedQueryFailoverSuite(t *testing.T) {
+	var querySuite *preparedQueryFailoverSuite
+	t.Parallel()
+	ct := NewCommonTopo(t)
+
+	for _, s := range preparedQueryFailoverSuites {
+		s.setup(t, ct)
+		querySuite = s
+	}
+	// setup dc3 cluster and disable server node in dc2
+	querySuite.setupForDC3(ct, ct.DC3, ct.DC1, ct.DC2)
```

Comment:
Shouldn't you run this for every suite? Currently only running on whatever the last one is.

Reply:
It should run for every suite. Not sure why it's only running at the end